wrappers/python/nntile/model/generation/llm_samplers.py

```diff
@@ -0,0 +1,99 @@
+# @copyright (c) 2022-present Skolkovo Institute of Science and Technology
+#                              (Skoltech), Russia. All rights reserved.
+#                2023-present Artificial Intelligence Research Institute
+#                              (AIRI), Russia. All rights reserved.
+#
+# NNTile is software framework for fast training of big neural networks on
+# distributed-memory heterogeneous systems based on StarPU runtime system.
+#
+# @file wrappers/python/nntile/model/generation/llm_samplers.py
+#
+# @version 1.1.0
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+from scipy.special import softmax
+
+from nntile.model.generation.llm_params import GenerationMode
+
+
+def sample_topk(logits, k, temperature):
+    batch_extended = logits if len(logits.shape) > 1 else logits[:, None]
```

Comment:
Further code relies on `len(logits.shape) == 2`, as it accesses `logins.shape[1]`. Maybe just add a corresponding assert? Then, this line will be no needed.

Reply:
made asserts
but probably batched and unbatched variations at the same time are good

Reply:
I meant your code accesses element `logits.shape[1]` which is a runtime error in case `len(logits.shape) == 1`. Assert if len of logits shape is equal 2, not less or equal.

Reply:
Alternatively, access `batch_extended.shape[1]` instead of `logits.shape[1]`